### ISSUE TITLE
Host documentation

### ISSUE THREAD

**Issue by kvark** [gfx-rs/metal-rs#5]
Most of the project rely on `docs.rs` for documentation, however it only builds Linux at the moment. We need to find a good way of providing the documentation here.

cc @fkaa @JohnColanduoni

**Comment by kvark** [comment]
Related - https://github.com/onur/docs.rs/pull/73

**Comment by mjadczak** [comment]
Also related to this: https://github.com/onur/docs.rs/issues/93

**Comment by IsaccBarker** [comment]
I know this is a really old issue, but I found a way to do it with GitHub's new Actions to Pages functionality. Below is an example, and you can find [working documentation hosted on my fork.](https://github.com/IsaccBarker/metal-rs):

```yaml
name: Rust

on:
  push:
    branches: [ "master" ]
  pull_request:
    branches: [ "master" ]
  workflow_dispatch:

permissions:
  contents: read
  pages: write
  id-token: write

concurrency:
  group: "pages"
  cancel-in-progress: true

env:
  CARGO_TERM_COLOR: always

jobs:
  build:
    runs-on: macos-latest
    steps:
    - uses: actions/checkout@v3
    - name: Document
      run: cargo doc --verbose
#    - name: Packaging
#      run: tar -zcvf github-pages.tar.gz ./target/doc
    - name: Packaging
      run: mv target/doc _site
    - name: Upload artifact
      uses: actions/upload-pages-artifact@v1
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
```
You have to set it up in the repository settings though.

**Comment by notgull** [comment]
> Most of the project rely on docs.rs for documentation, however it only builds Linux at the moment. 

Since this is no longer true, should this issue be closed?

**Comment by IsaccBarker** [comment]
If this still true? The [about builds page for docs.rs](https://docs.rs/about/builds) says:
> All targets other than x86_64-unknown-linux-gnu are cross-compiled. For implementation reasons, this is unlikely to change for the foreseeable future. 

Correct me if I'm wrong, but wouldn't out-of-ecosystem platform specific libraries still be needed, that couldn't be brought in?

**Comment by kvark** [comment]
@IsaccBarker would you mind making a PR?
What do I need to set up?

**Comment by madsmtm** [comment]
I'm able to generate documentation for [`icrate::Metal`](https://docs.rs/icrate/latest/icrate/Metal/index.html), I think the problem is mostly just that you've enabled `objc`'s `objc_exception` feature, which requires compiling with `cc`.

**Comment by cwfitzgerald** [comment]
I think this can be closed now that #303 landed
